src/inmanta/agent/agent.py

```diff
@@ -1039,37 +1057,90 @@ def __init__(
                 for name in names:
                     if "$" in name:
                         name = name.replace("$node-name", self.node_name)
-
-                    self.add_end_point_name(name)
+                    await self.add_end_point_name(name)
 
     async def stop(self) -> None:
         await super(Agent, self).stop()
         self.thread_pool.shutdown(wait=False)
         for instance in self._instances.values():
             await instance.stop()
 
+    async def do_start(self):
+        """
+            This method is required because:
+                1) The client transport is required to retrieve the autostart_agent_map from the server.
+                2) _init_endpoint_names() needs to be an async method and async calls are not possible in a constructor.
+        """
+        async with self._instances_lock:
+            await self._init_agent_map()
+            await self._init_endpoint_names()
+
     async def start(self) -> None:
         # cache reference to THIS ioloop for handlers to push requests on it
         self._io_loop = ioloop.IOLoop.current()
         await super(Agent, self).start()
 
-    def add_end_point_name(self, name: str) -> None:
-        SessionEndpoint.add_end_point_name(self, name)
+    async def add_end_point_name(self, name: str) -> None:
```

Comment:
Why is this async?

Reply:
To stay consistent with the `remove_end_point_name()` method which should be async.